### PR TITLE
Fix the signature for `onSync` for @types/workbox-background-sync

### PR DESCRIPTION
**Pull request by ghost**
The signature for `onSync` is incorrect. It should contain an argument containing the `queue` property.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/GoogleChrome/workbox/blob/v4.3.0/packages/workbox-background-sync/Queue.mjs#L39.

### CODE REVIEW

File: types/workbox-background-sync/Queue.d.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export class Queue {
 
 export interface QueueOptions {
     maxRetentionTime?: number;
-    onSync?: () => void;
+    onSync?: (options: { queue: Queue }) => void|Promise<void>;
```
Comment (JasonHK) [inline types/workbox-background-sync/Queue.d.ts:14]: I suggest extract the callback as a new type like this:

```typescript
interface QueueOnSyncCallback {
    (event: { queue: Queue }): void | Promise<void>;
}
```

Then change `QueueOptions.onSync` to this:

```suggestion
    onSync?: QueueOnSyncCallback;
```

**Review by JasonHK — CHANGES_REQUESTED** [review]

File: types/workbox-background-sync/Queue.d.ts [hunk 2]
```diff
@@ -9,9 +9,17 @@ export class Queue {
     unshiftRequest(entry: QueueEntry): Promise<void>;
 }
 
+export interface OnSyncCallbackOptions {
```
Comment (JasonHK) [inline types/workbox-background-sync/Queue.d.ts:12]:
```suggestion
export interface QueueOnSyncEvent {
```

File: types/workbox-background-sync/Queue.d.ts [hunk 3]
```diff
@@ -9,9 +9,17 @@ export class Queue {
...
+  queue: Queue;
+}
+
+export interface OnSyncCallback {
```
Comment (JasonHK) [inline types/workbox-background-sync/Queue.d.ts:16]:
```suggestion
export interface QueueOnSyncHandler {
```

File: types/workbox-background-sync/Queue.d.ts [hunk 4]
```diff
@@ -9,9 +9,17 @@ export class Queue {
...
+  (options: OnSyncCallbackOptions): void|Promise<void>;
```
Comment (JasonHK) [inline types/workbox-background-sync/Queue.d.ts:17]:
```suggestion
  (event: QueueOnSyncEvent): void | Promise<void>;
```

File: types/workbox-background-sync/Queue.d.ts [hunk 5]
```diff
@@ -9,9 +9,17 @@ export class Queue {
...
+}
+
 export interface QueueOptions {
     maxRetentionTime?: number;
-    onSync?: () => void;
+    onSync?: OnSyncCallback;
```
Comment (JasonHK) [inline types/workbox-background-sync/Queue.d.ts:22]:
```suggestion
    onSync?: QueueOnSyncHandler;
```

**Review by JasonHK — CHANGES_REQUESTED** [review]
Sorry, but I think that is an event, not an "option".

**Comment by owcs-solx** [comment]
> Sorry, but I think that is an event, not an "option".

I have no issues with naming it "event" . I am just following how Google is naming them at https://github.com/GoogleChrome/workbox/blob/master/packages/workbox-background-sync/src/Queue.ts.

I'll change them as per your suggestion.

**Review by JasonHK — APPROVED** [review]
Thank you! :smiley:

**Pull request merged**